operator/operator.php

```diff
@@ -64,12 +74,36 @@ public function get_user_categories()
 	 */
 	public function set_user_categories($forum_id)
 	{
-		// Set the collapsed category data array
-		$this->set_collapsed_categories($forum_id);
+		// Do not apply the ext functionality for bots
+		if (!$this->user->data['is_bot'])
```

Comment:
The bot protection is not needed here. In fact what we should do is when we switch over to template events (pending the event being added to the core) we should check if is a bot in the template event, and not display the buttons/links at all there instead :smile: I will add that note to the issue for that.

Also, just FYI, if we were to do something like this, rather than wrapping the whole method in a condition, I prefer this style:
```
if (something went wrong)
{
    return false;
}

// continue with normal function design and return values...